ui_framework/components/button/button.js

```diff
@@ -0,0 +1,147 @@
+import React, {
+  PropTypes,
+} from 'react';
+
+import classNames from 'classnames';
+import keyMirror from 'keymirror';
+
+import { KuiButtonIcon } from './button_icon/button_icon';
+
+const commonPropTypes = {
+  type: PropTypes.string,
+  testSubject: PropTypes.string,
```

Comment:
Hm, this could potentially be handled in the same way as I describe for "extra fields" (https://github.com/elastic/kibana/pull/10646#discussion_r107442633). Then you don't even have to include this here (I expect this is something we want to set in many places, and some of those places will be raw DOM elements, so there you'd have to write `data-test-subj` yourself instead of having it "in the abstraction)